apple/internal/ios_rules.bzl

```diff
@@ -2518,6 +2519,7 @@ ios_application = rule_factory.create_apple_rule(
         rule_attrs.app_icon_attrs(
             icon_extension = ".appiconset",
             icon_parent_extension = ".xcassets",
+            supports_alternate_icons = True,
```

Comment:
[question] Is this safe and backward compatible?

Reply:
It gotta be. I assume it wouldn't have landed in [upstream](https://github.com/bazelbuild/rules_apple/tree/upstream) otherwise.

Btw, this is a clean cherry-pick of the original commit [6fc1952](https://github.com/bazelbuild/rules_apple/commit/6fc19520455d0e48dd616a2ca8ddc6506d28f0f7), with only formatting and docs being changed from my side (through automation).